internal/services/storage/storage_account_resource_test.go

```diff
@@ -4357,6 +4372,31 @@ resource "azurerm_storage_account" "test" {
 `, data.RandomInteger, data.Locations.Primary, data.RandomString)
 }
 
+func (r StorageAccountResource) emptySharePropertiesPremiumFileStorage(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-storage-%[1]d"
+  location = "%[2]s"
+}
+
+resource "azurerm_storage_account" "test" {
+  name                = "unlikely23exst2acct%[3]s"
+  resource_group_name = azurerm_resource_group.test.name
+
+  location                 = azurerm_resource_group.test.location
+  account_tier             = "Premium"
+  account_replication_type = "LRS"
+  account_kind             = "FileStorage"
+
+  share_properties {}
```

Comment:
in this case this configuration shouldn't be valid, if the `share_properties` block is specified then a field within it should be (or the block should be omitted) - can we remove this (and update the schema to ensure that at least one of the items within this block is set?)

Reply:
Adding the `AtLeastOneOf` will breaks the existing test case: https://github.com/hashicorp/terraform-provider-azurerm/pull/21226/files#diff-4f9906e7fa788bca7bf275f811f2bb83e477040899f0da7b51ee414c8175340dL4355, where we already has supported the empty `share_properties`. This will then introduce a breaking change.